packages/instruction-plans/src/__tests__/instruction-plan-test.ts

```diff
@@ -1,7 +1,301 @@
 import '@solana/test-matchers/toBeFrozenObject';
 
-describe('InstructionPlan', () => {
-    it('will have tests #trust', () => {
-        expect(true).toBe(true);
+import { Address } from '@solana/addresses';
+import { pipe } from '@solana/functional';
+import { Instruction } from '@solana/instructions';
+import {
+    BaseTransactionMessage,
+    createTransactionMessage,
+    setTransactionMessageFeePayer,
+    TransactionMessageWithFeePayer,
+} from '@solana/transaction-messages';
+import { getTransactionMessageSize, TRANSACTION_SIZE_LIMIT } from '@solana/transactions';
+
+import {
+    getLinearMessagePackerInstructionPlan,
+    getMessagePackerInstructionPlanFromInstructions,
+    getReallocMessagePackerInstructionPlan,
+    nonDivisibleSequentialInstructionPlan,
+    parallelInstructionPlan,
+    sequentialInstructionPlan,
+    singleInstructionPlan,
+} from '../instruction-plan';
+
+jest.mock('@solana/transactions', () => ({
+    ...jest.requireActual('@solana/transactions'),
+    getTransactionMessageSize: jest.fn(),
+}));
+
+function createInstruction<TId extends string>(id: TId): Instruction & { id: TId } {
+    return { id, programAddress: '11111111111111111111111111111111' as Address };
+}
+
+describe('singleInstructionPlan', () => {
+    it('creates SingleInstructionPlan objects', () => {
+        const instruction = createInstruction('A');
+        const plan = singleInstructionPlan(instruction);
+        expect(plan).toEqual({ instruction, kind: 'single' });
+    });
+    it('freezes created SingleInstructionPlan objects', () => {
+        const instruction = createInstruction('A');
+        const plan = singleInstructionPlan(instruction);
+        expect(plan).toBeFrozenObject();
+    });
+});
+
+describe('parallelInstructionPlan', () => {
+    it('creates ParallelInstructionPlan objects from other plans', () => {
+        const instructionA = createInstruction('A');
+        const instructionB = createInstruction('B');
+        const plan = parallelInstructionPlan([
+            singleInstructionPlan(instructionA),
+            singleInstructionPlan(instructionB),
+        ]);
+        expect(plan).toEqual({
```

Comment:
All of these tests appear to pass using `toStrictEqual()` as well. Is it important to leave them open to _extra_ properties appearing without breaking a test, or is it important to defend against extra properties sneaking in?

Reply:
Good catch, this is just a mistake on my side. This should be `toStrictEqual`. 🙏